service/src/main/scala/za/co/absa/loginsvc/rest/AuthManagerConfig.scala

```diff
@@ -18,36 +18,49 @@ package za.co.absa.loginsvc.rest
 
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.context.annotation.{Bean, Configuration}
-import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.{AuthenticationManager, AuthenticationProvider}
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
 import org.springframework.security.config.annotation.web.builders.HttpSecurity
-import za.co.absa.loginsvc.rest.config.{ActiveDirectoryLDAPConfig, UsersConfig}
+import za.co.absa.loginsvc.rest.config.auth.{ActiveDirectoryLDAPConfig, UsersConfig}
 import za.co.absa.loginsvc.rest.provider.ConfigUsersAuthenticationProvider
 import za.co.absa.loginsvc.rest.provider.ad.ldap.ActiveDirectoryLDAPAuthenticationProvider
 
 @Configuration
-class AuthManagerConfig @Autowired()(
-  // TODO make it autowired but only if in config AD LDAP provider is set up (#28)
-  usersConfig: UsersConfig,
-  adLDAPConfig: ActiveDirectoryLDAPConfig
-){
+class AuthManagerConfig{
+
+  @Autowired(required = false)
+  private var usersConfig: UsersConfig = _
+  @Autowired (required = false)
+  private var adLDAPConfig: ActiveDirectoryLDAPConfig = _
```

Comment:
why vars?

I'd guess
```scala
  @Autowired(required = false)
  val usersConfig: UsersConfig = null
  @Autowired(required = false)
  val adLDAPConfig: ActiveDirectoryLDAPConfig = null
```
could be used.

Reply:
Implemented